### DISCUSSION TITLE
Is it possible to develop theia extension using java language?

### DISCUSSION THREAD

**Discussion by gommy15** [eclipse-theia/theia#12078] [General] [answered]
I want to develop theia-extension using Java language.
Is it possible to develop theia extension using java language?

**Comment by vince-fugnitto** [comment, marked as answer]
@gommy15 thank you for the discussion, but unfortunately no it is not possible.

The only use-case it could be possible today is to spawn a Java child process from our NodeJS backend, and you'll need to handle the JSON-RPC messages between the frontend and process yourself.

**Reply by gommy15** [reply]
Thank you.
